client.go

```diff
@@ -67,7 +67,44 @@ func WithOnClose(onClose func()) ClientOpts {
 // WithUnaryClientInterceptor sets the provided client interceptor
 func WithUnaryClientInterceptor(i UnaryClientInterceptor) ClientOpts {
 	return func(c *Client) {
-		c.interceptor = i
+		if c.interceptor == nil {
+			c.interceptor = i
+		} else {
+			WithChainUnaryClientInterceptor(i)(c)
+		}
+	}
+}
+
+// WithChainUnaryClientInterceptor sets the provided chain of client interceptors
+func WithChainUnaryClientInterceptor(interceptors ...UnaryClientInterceptor) ClientOpts {
+	return func(c *Client) {
+		if c.interceptor != nil {
+			interceptors = append([]UnaryClientInterceptor{c.interceptor}, interceptors...)
+		}
+		c.interceptor = func(
+			ctx context.Context,
+			req *Request,
+			reply *Response,
+			info *UnaryClientInfo,
+			final Invoker,
+		) error {
+			return interceptors[0](ctx, req, reply, info,
```

Comment:
what if the `interceptors` is empty slice.

Reply:
You mean if a client is created with an empty `WithChainUnaryClientInterceptor()` option ? I added a length check to protect against that.